tests/xharness/AppRunner.cs

```diff
@@ -747,7 +749,7 @@ public async Task<int> RunAsync ()
 			if (!success.Value) {
 				int pid = 0;
 				string crash_reason = null;
-				foreach (var crash in crash_reports.Logs) {
+				foreach (var crash in Logs) {
```

Comment:
This will make xharness process every log as if it was a crash report, and that's quite wasteful, some of the logs can be quite big. It's better to just process logs we know might be crash reports.

Reply:
I was thinking the same when writing this  but my refactor uncovered possible problems like these because if you look at the original:
1. `Logs` is passed to the ctor above on line 434
2. I was able to make `Logs` readonly inside `CrashReportSnapshot` because it is only assigned in ctor so it never changes
3. Then we read the value back here but it's the same `Logs` we pass in so I didn't change anything and we already do what you are saying. Or I am wrong somehow but I let the code show me this while refactoring

Reply:
The Logs field/property itself may not have changed, but there can have been any number of Log instances added to the Logs instance after it was created.

Reply:
I understand that. But do you want me to fix it or leave it working like it is (was)?

Reply:
Please fix it to only iterate over actual crash reports instead of every log.

Reply:
Fixed